main.cpp

```diff
@@ -43,7 +42,16 @@ int main()
 
     InterruptIn btn(DEMO_BUTTON);
 
-    tr_info("Hello version %s", version);
+    // Get the current version from the mcuboot header information
+    struct image_version version;
+    ret = boot_get_current_version(&version);
+    if(ret == 0) {
+        tr_info("Hello version %d.%d.%d+%lu", version.iv_major, version.iv_minor,
+                                              version.iv_revision, version.iv_build_num);
+    } else {
+        tr_error("Failed to load version information: %d", ret);
+    }
```

Comment:
Good idea!

Reply:
Yes, I had to add that API to mcuboot... see mcu-tools/mcuboot#863

I'm hoping they accept the new function into the API.